apps/dashboard/src/components/editor/document-settings.js

```diff
@@ -53,7 +51,7 @@ const DocumentSettings = ( { project, onChangeThemeClick } ) => {
 			return;
 		}
 
-		saveEditorChanges( { public: true } );
+		saveAndUpdateProject( project.id, { public: true } );
```

Comment:
Having looked at D78067-code, it seems to me like this would re-publish the last-published version.  
So if you've made your project private, made some changes, and hit 'public' in the visibility menu again, it'd publish the old version.

It does kind of make sense logically... that said, you have no reference for what is actually being published when you perform the action which is kind of bad. Hence I have to ask if that's intentional or if I'm missing anything?

Reply:
Yes, this was intentional. The reason I'm suggesting this change is to avoid putting the user in a 'no turning back' situation. Which will happen silently on our live version.
I'm also using the `publicContent` contents to validate if the project was published at some point and then show the notification when the user hits Publish/Update.

As an alternative, we could keep this logic as in the current live version but show another notification (the message should be different) warning the user that they wouldn't be able to restore their public version when changing from public to private.
But, to use this approach, we'd also need to create some flag to let us know that the project was published at some point, thus, allowing us to show the warning notification when necessary.